src/render.js

```diff
@@ -1,72 +1,86 @@
-      }.bind(this)).map(geojson => geojson.properties.id);
+      return {source: 'cold', 'id': id};
     }
+  }).forEach(function calculateViewUpdate(change) {
+    let {id, source} = change;
+    let feature = store.get(id);
+    let featureInternal = feature.internal(mode);
 
-    this.sources.hot = [];
-    let lastColdCount = this.sources.cold.length;
-    this.sources.cold = this.isDirty ? [] : this.sources.cold.filter(function saveColdFeatures(geojson) {
-      var id = geojson.properties.id || geojson.properties.parent;
-      return newHotIds.indexOf(id) === -1;
+    store.ctxevents.currentModeRender(featureInternal, function addGeoJsonToView(geojson) {
+      store.sources[source].push(geojson);
     });
+  });
 
-    var coldChanged = lastColdCount !== this.sources.cold.length || newColdIds.length > 0;
-
-    newHotIds.concat(newColdIds).map(function prepForViewUpdates(id) {
-      if (newHotIds.indexOf(id) > -1) {
-        return {source: 'hot', 'id': id};
-      }
-      else {
-        return {source: 'cold', 'id': id};
-      }
-    }).forEach(function calculateViewUpdate(change) {
-      let {id, source} = change;
-      let feature = this.get(id);
-      let featureInternal = feature.internal(mode);
+  if (coldChanged) {
+    store.ctxmap.getSource('mapbox-gl-draw-cold').setData({
+      type: 'FeatureCollection',
+      features: store.sources.cold
+    });
+  }
 
-      this.ctx.events.currentModeRender(featureInternal, function addGeoJsonToView(geojson) {
-        this.sources[source].push(geojson);
-      }.bind(this));
-    }.bind(this));
+  store.ctxmap.getSource('mapbox-gl-draw-hot').setData({
+    type: 'FeatureCollection',
+    features: store.sources.hot
+  });
 
-    if (coldChanged) {
-      this.ctx.map.getSource('mapbox-gl-draw-cold').setData({
-        type: 'FeatureCollection',
-        features: this.sources.cold
-      });
-    }
+  let changed = store.getChangedIds().map(id => store.get(id))
+    .filter(feature => feature !== undefined)
+    .filter(feature => feature.isValid())
+    .map(feature => feature.toGeoJSON());
 
-    this.ctx.map.getSource('mapbox-gl-draw-hot').setData({
-      type: 'FeatureCollection',
-      features: this.sources.hot
-    });
+  if (changed.length) {
+    store.ctxmap.fire('draw.changed', {features: changed});
+  }
 
-    let changed = this.getChangedIds().map(id => this.get(id))
-      .filter(feature => feature !== undefined)
-      .filter(feature => feature.isValid())
-      .map(feature => feature.toGeoJSON());
+  const flushedSelectionSets = store.flushSeleted();
+  if (flushedSelectionSets.selected.length) {
+    store.ctxmap.fire('draw.select', { featureIds: flushedSelectionSets.selected });
+  }
+  if (flushedSelectionSets.deselected.length) {
+    store.ctxmap.fire('draw.deselect', { featureIds: flushedSelectionSets.deselected });
+  }
```

Comment:
https://github.com/mapbox/mapbox-gl-draw/pull/368/files/1cd2f63e2eabaa74f230cad147335133e9838223#diff-3bf35cd41da51153dcca3003656f7c67R192

I keep track of selected and deselected items since the last flush, then return them when you run that function.